### PR TITLE
[Keyboard] Add Tyson60s keyboard

### PR DESCRIPTION
**Pull request by trnthsn**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/trnthsn/tyson60s/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+     * │ Shift  │ Z │ X │ C │ V │ B │ N │ M │ , │ . │ / │ Shift│Fn │
+     * ├─────┬──┴┬──┴──┬┴───┴───┴───┴───┴───┴───┴──┬┴───┴┬───┬─┴───┤
+     * │Ctrl │GUI│ Alt │                           │ Alt │GUI│ Ctrl│
+     * └─────┴───┴─────┴───────────────────────────┴─────┴───┴─────┘
+     */
+    [0] = LAYOUT_all(
+        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_GRAVE, KC_BSPC,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
+        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, MO(1),
```
Comment (waffle87) [inline keyboards/trnthsn/tyson60s/keymaps/default/keymap.c:36]: This goes out of bounds.
```suggestion
        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, _______,
```

File: keyboards/trnthsn/tyson60s/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+     * │ Shift  │ Z │ X │ C │ V │ B │ N │ M │ , │ . │ / │ Shift│Fn │
+     * ├─────┬──┴┬──┴──┬┴───┴───┴───┴───┴───┴───┴──┬┴───┴┬───┬─┴───┤
+     * │Ctrl │GUI│ Alt │                           │ Alt │GUI│ Ctrl│
+     * └─────┴───┴─────┴───────────────────────────┴─────┴───┴─────┘
+     */
+    [0] = LAYOUT_all(
+        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_GRAVE, KC_BSPC,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
+        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, MO(1),
```
Comment (waffle87) [inline keyboards/trnthsn/tyson60s/keymaps/via/keymap.c:36]:
```suggestion
        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, _______,
```

**Review by waffle87 — APPROVED** [review]

**Comment by trnthsn** [comment]
thank you @waffle87

File: keyboards/trnthsn/tyson60s/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+     * └─────┴───┴─────┴───────────────────────────┴─────┴───┴─────┘
+     */
+    [0] = LAYOUT_all(
+        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_GRAVE, KC_BSPC,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,
+        KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, MO(1),
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, KC_RGUI, KC_RCTL
+    )
+
```
Comment (zvecr) [inline keyboards/trnthsn/tyson60s/keymaps/default/keymap.c:39]:
```suggestion
    ),
    [1] = LAYOUT_all(
        QK_BOOT, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______,                            _______,                            _______, _______, _______
    )
```

File: keyboards/trnthsn/tyson60s/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+     * └─────┴───┴─────┴───────────────────────────┴─────┴───┴─────┘
+     */
+    [0] = LAYOUT_all(
+        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_GRAVE, KC_BSPC,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,
+        KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, MO(1),
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, KC_RGUI, KC_RCTL
+    )
+
```
Comment (zvecr) [inline keyboards/trnthsn/tyson60s/keymaps/via/keymap.c:39]:
```suggestion
    ),
    [1] = LAYOUT_all(
        QK_BOOT, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______,                            _______,                            _______, _______, _______
    )
```

**Review by zvecr — CHANGES_REQUESTED** [review]
`MO(1)` for the keymaps would go out of bounds.

File: keyboards/trnthsn/tyson60s/keymaps/default/keymap.c (hunk 3 above)
Comment (trnthsn) [reply keyboards/trnthsn/tyson60s/keymaps/default/keymap.c]: i just modify 2nd and 3rd layer layout

Comment (trnthsn) [reply keyboards/trnthsn/tyson60s/keymaps/default/keymap.c]: i has pushed new 2nd and 3rd layer layout

File: keyboards/trnthsn/tyson60s/keymaps/via/keymap.c (hunk 4 above)
Comment (trnthsn) [reply keyboards/trnthsn/tyson60s/keymaps/via/keymap.c]: i has pushed new 2nd and 3rd layer layout

File: keyboards/trnthsn/tyson60s/keymaps/default/keymap.c (hunk 3 above)
Comment (trnthsn) [reply keyboards/trnthsn/tyson60s/keymaps/default/keymap.c]: @zvecr can you re-review my pull request

**Comment by trnthsn** [comment]
@zvecr can you review my pull request again

**Comment by zvecr** [comment]
As of August 26, 2024, `qmk/qmk_firmware` is no longer accepting VIA-enabled keymaps as these have now transitioned to a repository under the VIA team's control.

As you've submitted a PR containing `via` or VIA-enabled keymap(s), this is your notice that they should be removed from this PR. You should now submit a secondary PR to the [VIA QMK Userspace](https://github.com/the-via/qmk_userspace_via) repository with your associated `via` or VIA-enabled keymaps instead.

**Comment by trnthsn** [comment]
> As of August 26, 2024, `qmk/qmk_firmware` is no longer accepting VIA-enabled keymaps as these have now transitioned to a repository under the VIA team's control.
> 
> As you've submitted a PR containing `via` or VIA-enabled keymap(s), this is your notice that they should be removed from this PR. You should now submit a secondary PR to the [VIA QMK Userspace](https://github.com/the-via/qmk_userspace_via) repository with your associated `via` or VIA-enabled keymaps instead.

image
I has removed via layout folder a few hours ago, can you review my PR again

**Review by zvecr — APPROVED** [review]

File: keyboards/trnthsn/tyson60s/keyboard.json [hunk 5]
```diff
@@ -0,0 +1,400 @@
...
+    "ws2812": {
+        "driver": "vendor",
+        "pin": "GP29"
+    },
+    "layout_aliases": {
+        "LAYOUT_all": "LAYOUT_60_tsangan_hhkb"
+    },
+    "community_layouts": [
+        "60_ansi_tsangan",
+        "60_tsangan_hhkb",
```
Comment (fauxpark) [inline keyboards/trnthsn/tyson60s/keyboard.json:53]: Just a heads up, `60_tsangan_hhkb` will be deprecated in favour of `60_ansi_tsangan_split_bs_rshift` as of #24186.

**Pull request merged**